### PR TITLE
Refactoring all components that used StatusChip to use the new ones.

### PR DESCRIPTION
**Pull request by denektenina**
# Description

This is the second PR that was made as a solution to two problems that became evident when I needed to use the existing StatusChip with a new icon. 

**The first PR solved the following problems:**
1) The existing StatusChip accepts a string representing the icon as a prop, meaning that if I need to use a new icon I have to modify StatusChip to accept a prop for the new icon. This violates the Open / Closed Principe and can cause an unnecessary messy component over time, since every icon that might be used only one places has to be a prop and a new if/else in the component.

2) The component takes props for icon, text, color and padding, meaning that even though we use the same component, they might look very different across the app, losing some of the point of using the same component.

-> To solve the first problem I propose modifying the StatusChip component to accept icon as `children` instead so that new icons can be used without having to modify the component each time. (New name: BaseStatusChip)

-> To solve the second problem I propose adding a new component (new StatusChip) that hard codes a number of base styles, so that the choice of color and icon for similar states is consolidated across the app. Because the use case can be very different across the app, I am leaving the text as a prop still, even though the styling is the same. I.e. in some cases something might be "ongoing" but in other cases it might be "pending" - but they both use a light blue chip with an hourglass.

StatusChip will use BaseStatusChip to ensure the same styling.

BaseStatusChip now adheres to the Open / Closed Principe. StatusChip does not, but BaseStatusChip can be used in the edge cases not covered in StatusChip, ensuring that it will rarely or never need to be modified.

**This second PR uses the new components instead of the old, and deletes the old component.**

# How to test

There are three types of cases where the old `StatusChip` component was used:
* Inside `TicketStatusChip`: Here, we extracted the component into its own file, and refactored to use the new StatusChip instead of the old. All components complied with the base styles, so there shouldn't be any problems here, but if you want to be thorough you can check every view that renders a TicketStatusChip to check that it looks good.
* Inside `NviStatusChip`: Same as above
* Some independent uses of `StatusChip`: I think there were 4 such cases. Two of them corresponded with the base styles and could be easily switched, two diverged. One by a slightly different color, one by removing y-padding. These styles were preserved by using `BaseStatusChip` instead, but it could be considered if the difference is so important to preserve that it's necessary. I'm saving this question for another day, since the goal of this PR is to change no rendering logic / UI.

# Checklist

Ensure that the changes are aligned with the expectations of PO/designer before marking the PR as ready for review.

Also ensure that the following criterias are met:

- [x] The changes are working as expected
- [x] The changes are tested OK for different screen sizes
- [x] The changes are tested OK for a11y
- [x] Interactive elements have data-testids
- [x] I have done a QA of my own changes

_Note: some of these criterias may not always be relevant, and can simply be marked as completed._

### CODE REVIEW

File: src/pages/messages/components/TicketListItem.tsx [hunk 1]
```diff
@@ -117,9 +120,12 @@ export const TicketListItem = ({ ticket }: TicketListItemProps) => {
...
           {ticket.type === 'GeneralSupportCase' && isOnMyPageMessages ? (
             viewedByUser ? (
-              <StatusChip text={t('common.read_past_tense')} icon="check" />
+              <BaseStatusChip sx={{ bgcolor: 'info.light' }}>
+                <CheckIcon aria-hidden="true" fontSize={'small'} />
+                <Typography>{t('common.read_past_tense')}</Typography>
+              </BaseStatusChip>
             ) : (
-              <StatusChip text={t('common.unread')} icon="hourglass" />
+              <StatusChip status={StatusValue.InProgress} text={t('common.unread')} />
```
[the remark this answers is not in the document]
Comment (denektenina) [reply src/pages/messages/components/TicketListItem.tsx:128]: We are trying to use a limited amount of base states with different text. So this is similar enough to be good.

**Review by iBuzza — APPROVED** [review]
